### PR TITLE
Docs: Add `Streaming upsert write`  in flink.md

### PR DESCRIPTION
**Pull request by xiaotianzhang01**
Add a separate use case of `write.upsert.enabled` in the flink document to answer some questions of flink users. Is this necessary?

### CODE REVIEW

File: site/docs/flink.md [hunk 1]
```diff
@@ -522,6 +522,18 @@ FlinkSink.forRowData(input)
 env.execute("Test Iceberg DataStream");
 ```
 
+## Streaming upsert write
+
+Iceberg supports upsert based on the primary key when writing streaming data.
```
Comment (samredai) [inline site/docs/flink.md:527]: Would it be helpful to state here that this requires v2 tables?

Comment (xiaotianzhang01) [reply site/docs/flink.md:527]: Thank you for your reply, this can be removed. The introduction of this feature is needed for new users of iceberg

Comment (xiaotianzhang01) [reply site/docs/flink.md]: yeap,  update

File: site/docs/flink.md [hunk 2]
```diff
@@ -522,6 +522,20 @@ FlinkSink.forRowData(input)
...
+Iceberg supports upsert based on the primary key when writing streaming data into v2 table format.
+
+```sql
+CREATE TABLE `hive_catalog`.`default`.`sample` (
+  `id`  INT UNIQUE COMMENT 'unique id',
+  `data` STRING NOT NULL,
+ PRIMARY KEY(`id`) NOT ENFORCED
+) with ('format-version'='2', 'write.upsert.enabled'='true');
+```
+!!! Note
```
Comment (Reo-LEI) [inline site/docs/flink.md:536]: In UPSERT mode, if the table is partitioned, the partition fields should be a part of equality fields. I think this is important and should be mentioned in the docs. 

https://github.com/apache/iceberg/blob/e0ff71dfa8ffb54941fcc82ae127ab06f271331c/flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java#L378

Comment (xiaotianzhang01) [reply site/docs/flink.md:536]: okey

**Comment by xiaotianzhang01** [comment]
CC: @openinx @Reo-LEI @rdblue @jackye1995 @RussellSpitzer

**Comment by Reo-LEI** [comment]
LGTM, Thanks for your contribution! @xiaotianzhang01

**Comment by linfan** [comment]
Why this PR still laying here ? I think someone should resolve and merge it...

**Comment by openinx** [comment]
Sorry, I think I've missed this PR before.  @xiaotianzhang01 , would you mind to fix those conflicts ?

**Pull request closed without merging**